Added timeout connection option

lzap opened the pull request:
Hello there! This adds timeout option parameter, to make it less verbose I propose only to add unified timeout parameter to set both timeouts at once:

* open timeout
* read timeout

Unfortunately write timeout (Ruby 2.6+ for Net::HTTP) is not supported at all. I only tested this with Net::HTTP honestly.

Edit: Actually also write timeout is set for Net::HTTP: https://github.com/lostisland/faraday/blob/d4f062a56b201af97ac1c0d3b9bcc0487b5dc408/lib/faraday/adapter/net_http.rb#L167-L183 and it looks like the two Faraday timeouts are "mapped" to multiple timeouts in various implementations so I think providing those in one global timeout should not be an issue.

lzap commented:
Travis failed, Bundler 1 vs 2 issue. If you need me to take a look, I can. Looks like there is not much activity going on here, I can also offer co-maintainship. I'd be very interested in pushing out new release with the timeouts as our customer needs this. Thanks!

ares commented:
good addition

lzap commented:
@ananace cheers, how you doin` ? ;-)

unclebilly commented:
Got some seemingly unrelated build failures here - I will take a look.

unclebilly commented:
Hello @lzap - apologies for the delay.  I fixed the Travis build issues, which were unrelated to this branch.  If you rebase from master, the build should pass on this branch and we can merge it.

lzap commented:
Hey, thanks so much. Added an integration test which only runs when INTEGRATION_TIMEOUT env variable is set (it opens up a socket, might break in some environments). The test does not run with 1.8 ruby where it fails - timeout seems to be 60 seconds there. I am assuming that Faraday gem is too old on that version. Shall we put a note in README that the timeout will only work on Ruby 1.9+ ?

unclebilly commented:
Yes, my hope is that we don't have any users with < 1.9.  I don't know
either way.  Please feel free to remove that version from the travis YAML
file and update the README as part of this merge!

On Thu, Jun 6, 2019 at 6:38 AM Lukáš Zapletal <notifications@github.com>
wrote:

> Hey, thanks so much. Added an integration test which only runs when
> INTEGRATION_TIMEOUT env variable is set (it opens up a socket, might break
> in some environments). The test does not run with 1.8 ruby where it fails -
> timeout seems to be 60 seconds there. I am assuming that Faraday gem is too
> old on that version. Shall we put a note in README that the timeout will
> only work on Ruby 1.9+ ?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/govdelivery/infoblox/pull/59?email_source=notifications&email_token=AAAXBHQ2HUDRHKG3NW266CDPZDZJTA5CNFSM4HR246FKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXCSIBY#issuecomment-499459079>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAXBHUIJ533WDYTQAREQPDPZDZJTANCNFSM4HR246FA>
> .
>

lzap commented:
Updated README, dropped 1.8.7 from travis, removed test skip. I think this is worth bumping version to `2.1` or even `3.0` so it won't break peoples CI jobs if they require the gem properly (on 1.8.x version I mean).

unclebilly commented on spec/integration/connection_timeout_spec.rb:
```diff
@@ -0,0 +1,32 @@
+if ENV['INTEGRATION_TIMEOUT']
```
Is this environment flag necessary anymore? Sorry, I wasn't clear earlier but I was thinking this environment flag would get removed after we pulled 1.8.7 support.

lzap replied:
My bad, indeed that makes sense. Rebased.

unclebilly commented:
@lzap Awesome, thanks!  I'll fix the build failure in 1.9.3 (again unrelated to this work) and cut a new version.

lzap commented:
Hey, what is the ETA please? I need to know to schedule some updates in Red Hat Satellite, we have multiple customers having problems with this and I either need to decide to wait for your release or push out this bug from planned errata. Many thanks for info!

lzap commented:
Also dropped the ENV variable from `.travis.yml` it was unused.

unclebilly commented:
I will push out a new release today; the issue with 1.9.3 is not related to this branch and I can fix it outside of this context.

unclebilly commented:
Merged - new gem will be cut shortly.

unclebilly commented:
@lzap verion 3.0.0 has been pushed to rubygems.org.  Thanks for your contribution (and patience).

lzap commented:
Thanks a bunch!
